source/MaterialXGenShader/Util.cpp

```diff
@@ -508,4 +509,105 @@ void mapValueToColor(const ValuePtr value, Color4& color)
     }
 }
 
+bool requiresImplementation(const NodeDefPtr nodeDef)
+{
+    if (!nodeDef)
+    {
+        return false;
+    }
+    static std::string TYPE_NONE("none");
+    const std::string typeAttribute = nodeDef->getAttribute(TypedElement::TYPE_ATTRIBUTE);
```

Comment:
Could be changed to `nodeDef->getType()`. Also make the string a reference to avoid string copy.